pyhf/commandline.py

```diff
@@ -0,0 +1,21 @@
+import logging
+logging.basicConfig()
+log = logging.getLogger(__name__)
+
+import click
+import json
+from . import readxml
+
+@click.group(context_settings=dict(help_option_names=['-h', '--help']))
+def pyhf():
+    pass
+
+@pyhf.command()
+@click.option('--entrypoint-xml', required=True, prompt='Top-level XML', help='The top-level XML file for the PDF definition.', type=click.Path(exists=True))
+@click.option('--basedir', required=True, prompt='Base directory', help='The base directory for the XML files to point relative to.', type=click.Path(exists=True))
+@click.option('--output-file', required=True, prompt='Output file', help='The location of the output json file. If not specified, prints to screen.')
```

Comment:
what do you think about making the `entrypoint-xml` be a `click.argument` there is really not way to convert without input, so `pyhf xml2json input.xml` seems to be a good cmd line

`--basedir` could default to `os.getcwd()`

also, maybe it's somewhat more unixy to print to stdout if the output file is not provided?

`pyhf xml2json input.xml > test.json` ?

Reply:
> pyhf xml2json input.xml > test.json ?

The good part is `tqdm` is part of `stderr` so we can definitely do that.